docs/design/api/storage.rst

```diff
@@ -0,0 +1,324 @@
+====================================================================
+
+You'll need to do a BATCH operation with all the sub-operations in there.
+
+- First get the list of records you want to modify.
+
+.. code-block:: http
+
+    GET /buckets/mozilla/collections/payments/records?seller=hawk:144c22a77d75937740ec3c957fbdb1d1
+    Authorization: Basic <PaymentApp Basic Auth credentials>
+    
+    HTTP/1.1 200 OK
+    {
+      "items": [
+        {
+            "id": "dc86afa9-a839-4ce1-ae02-3d538b75496f",
+            "last_modified": 1432024555580,
+            "buyer": "fxa:5331be33303ccff19d3c49b6da276913",
+            "seller": "hawk:144c22a77d75937740ec3c957fbdb1d1",
+            "product": "bb5bf35f-cb2b-40e7-b1ef-8b097bd550f2",
+            "added_on": 1432024555580,
+            "valid_until": 1463560555580
+        },
+        {
+            "id": "db4d95e1-c076-4848-950c-cf462b2631f0",
+            "last_modified": 1430140411480,
+            "title": "MoFo",
+            "buyer": "fxa:465afecea6b565c85fd980a603747fec",
+            "seller": "hawk:144c22a77d75937740ec3c957fbdb1d1",
+            "product": "ecd68f3c-984b-471c-a670-8411e5247358",
+            "added_on": 1430140411480,
+            "valid_until": 1432732411480
+        }
+      ]
+    }
+
+
+- Then run a BATCH requests.
+
+If you want to add the ``read`` permission for the seller app to all records of the app, you could use:
+
+.. code-block:: http
+
+    POST /batch
+    Authorization: Basic <PaymentApp Basic Auth credentials>
+    
+    {
+      "defaults": {
+      "method" : "PATCH",
+      "data": {
+        "permissions": {
+          "read": ["+hawk:144c22a77d75937740ec3c957fbdb1d1"]
+        }
+      }
+    },
+    "requests": [
+      {
+        "path" : "/buckets/mozilla/collections/payments/records/dc86afa9-a839-4ce1-ae02-3d538b75496f"
+      },
+      {
+        "path" : "/buckets/mozilla/collections/payments/records/db4d95e1-c076-4848-950c-cf462b2631f0"
+      }
+    ]
+  }
+
+As the selling application I want to be able to access the purchase information for agivenuser for my application
+=================================================================================================================
+
+.. code-block:: http
+
+    GET /buckets/mozilla/collections/payments/records?buyer=fxa:5331be33303ccff19d3c49b6da276913
+    Authorization: Hawk mac="kDPC...=", hash="B0we...=", id="144c22a77d75937740ec3c957fbdb1d1", ts="1432030137", nonce="mQao38"
+    
+    HTTP/1.1 200 OK
+    {
+      "items": [
+        {
+            "id": "dc86afa9-a839-4ce1-ae02-3d538b75496f",
+            "last_modified": 1432024555580,
+            "buyer": "fxa:5331be33303ccff19d3c49b6da276913",
+            "seller": "hawk:144c22a77d75937740ec3c957fbdb1d1",
+            "product": "bb5bf35f-cb2b-40e7-b1ef-8b097bd550f2",
+            "added_on": 1432024555580,
+            "valid_until": 1463560555580
+        }
+      ]
+    }
+
```

Comment:
IMO it would be worth mentionning here that the filtering on records where the current user has read access is implicit.